pyvista/core/utilities/points.py

```diff
@@ -459,3 +464,161 @@ def vector_poly_data(orig, vec):
+
+    The axes may be used to build an oriented bounding box or to align the points to
+    another set of axes (e.g. the world XYZ axes).
+
+    .. note::
+        The computed axes are not unique, and the sign of each axis direction can be
+        arbitrarily changed.
+
+    .. note::
+        This implementation creates a temporary array of the same size as the input
+        array, and is therefore not optimal in terms of its memory requirements.
+        A more memory-efficient computation may be supported in a future release.
+
+    .. versionadded:: 0.45.0
+
+    Parameters
+    ----------
+    points : MatrixLike[float]
+        Nx3 array of points.
+
+    return_std : bool, default: False
+        If ``True``, also returns the standard deviation of the points along each axis.
+        Standard deviation is computed as the square root of the eigenvalues of the
+        mean-centered covariance matrix.
+
+    Returns
+    -------
+    numpy.ndarray
+        3x3 orthonormal array with the principal axes as row vectors.
+
+    numpy.ndarray
+        Three-item array of the standard deviations along each axis.
+
+    Examples
+    --------
+    >>> import pyvista as pv
+    >>> import numpy as np
+    >>> rng = np.random.default_rng(seed=0)  # only seeding for the example
+
+    Create a mesh with points that have the largest variation in ``X``,
+    followed by ``Y``, then ``Z``.
+
+    >>> radii = np.array((6, 3, 1))  # x-y-z radii
+    >>> mesh = pv.ParametricEllipsoid(
+    ...     xradius=radii[0], yradius=radii[1], zradius=radii[2]
+    ... )
+
+    Plot the mesh and highlight its points in black.
+
+    >>> p = pv.Plotter()
+    >>> _ = p.add_mesh(mesh)
+    >>> _ = p.add_points(mesh, color='black')
+    >>> _ = p.show_grid()
+    >>> p.show()
+
+    Compute its principal axes and return the standard deviations.
+
+    >>> axes, std = pv.principal_axes(mesh.points, return_std=True)
+    >>> axes
+    pyvista_ndarray([[-1.0000000e+00, -3.8287229e-08,  3.6589407e-10],
+                     [-3.8287229e-08,  1.0000000e+00, -3.0685656e-09],
+                     [-3.6589393e-10, -3.0685656e-09, -1.0000000e+00]],
+                    dtype=float32)
+
+    Note that the principal axes have ones along the diagonal and zeros
+    in the off-diagonal. This indicates that the first principal axis is
+    aligned with the x-axis, the second with the y-axis, and third with
+    the z-axis. This is expected, since the mesh is already axis-aligned.
+
+    However, since the signs of the principal axes are arbitrary, the
+    first and third axes in this case have a negative direction.
+
+    Show the standard deviation along each axis.
+
+    >>> std
+    array([3.014956 , 1.507478 , 0.7035637], dtype=float32)
+
+    Compare this to using :meth:`numpy.std` for the computation.
+
+    >>> np.std(mesh.points, axis=0)
+    pyvista_ndarray([3.0149572, 1.5074761, 0.7035699], dtype=float32)
+
+    Convert the values to proportions for analysis.
```

Comment:
```suggestion
    Since the points are axis-aligned, the two results agree in this case. In general, 
    however, these two methods differ in that :meth:`numpy.std` with `axis=0` computes
    the standard deviation along the `x-y-z` axes, whereas the standard deviation 
    returned by :meth:`principal_axes` is computed along the principal axes.

    Convert the values to proportions for analysis.
```